Modules/Relationships/src/Relationships.Application/Infrastructure/Persistence/Repository/IRelationshipTemplatesRepository.cs

```diff
@@ -11,4 +12,6 @@ public interface IRelationshipTemplatesRepository
     Task<RelationshipTemplate> Find(RelationshipTemplateId id, IdentityAddress identityAddress, CancellationToken cancellationToken, bool track = false, bool fillContent = true);
     Task Add(RelationshipTemplate template, CancellationToken cancellationToken);
     Task Update(RelationshipTemplate template);
+    Task Delete(IEnumerable<RelationshipTemplateId> templateIds, CancellationToken cancellationToken);
+    Task DeleteTemplates(Expression<Func<RelationshipTemplate, bool>> filter, CancellationToken cancellationToken);
```

Comment:
Same as above, albeit this isn't a delete by id, I think the losses on code readability make it hard to justify this approach.
We don't even have multiple use cases with different filters, and typically when we do, we use QueryableExtensions.